### ISSUE TITLE
`avoid_positional_boolean_parameters` ignored when in function-only-parameter 

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#4349]
I'm proposing here that `avoid_positional_boolean_parameters` could be ignored in function-only-parameter cases.

## Bad Examples

```dart
Task(true);
Task(false);
ListBox(false, true, true);
Button(false);

//New Case:
Task.doSomething(object, true);
```

## Good Examples 

```dart
Task.oneShot();
Task.repeating();
ListBox(scroll: true, showScrollbars: true);
Button(ButtonState.enabled);

//New cases: 
object.select(true);
dataRow.onSelectionChanged(true);
```

## Discussion

I believe that when using boolean parameters as the only parameter in the function, is most likely a function that has a meaningful name for that value already. Similar to the ones I mentioned above in the actual Flutter repo (`onSelectionChanged`).

### Discussion checklist

- [x] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [ ] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [x] If there's any prior art (e.g., in other linters), please add references here.
- [x] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [x] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

**Comment by srawlins** [comment]
This is largely a duplicate of https://github.com/dart-lang/linter/issues/1638, sort of an alternative solution to that issue.
